lib/src/spec/common.dart

```diff
@@ -74,31 +75,189 @@ abstract class StatsMessageTraffic {
   StatsMessageTypes webhook;
 }
 
+/// A class providing parameters of a token request.
+///
+/// Parameters for a token request
+///
+/// [Auth.authorize], [Auth.requestToken] and [Auth.createTokenRequest]
+/// accept an instance of TokenParams as a parameter
+///
+/// spec: https://docs.ably.io/client-lib-development-guide/features/#TK1
 class TokenParams {
+
+  /// Capability of the token.
+  ///
+  /// If the token request is successful, the capability of the
+  /// returned token will be the intersection of this capability
+  /// with the capability of the issuing key.
+  ///
+  /// Ref: TK2b
   String  capability;
+
+  /// A clientId to associate with this token.
+  ///
+  /// The generated token may be used to authenticate as this clientId.
+  ///
+  /// Ref: TK2c
   String clientId;
+
+  /// An opaque nonce string of at least 16 characters to ensure uniqueness.
+  ///
+  /// Timestamps, in conjunction with the nonce,
+  /// are used to prevent requests from being replayed
+  ///
+  /// ref: TK2d
   String nonce;
+
+  /// The timestamp (in millis since the epoch) of this request.
+  ///
+  ///	Timestamps, in conjunction with the nonce, are used to prevent
+  ///	token requests from being replayed.
+  ///
+  /// ref: TK2d
   DateTime timestamp;
+
+  /// Requested time to live for the token.
+  ///
+  /// If the token request is successful, the TTL of the returned
+  /// token will be less than or equal to this value depending on
+  /// application settings and the attributes of the issuing key.
+  ///
+  /// 0 means Ably will set it to the default value
+  ///
+  /// Ref: TK2a
   int ttl;
+
+  TokenParams({
+    this.capability,
+    this.clientId,
+    this.nonce,
+    this.timestamp,
+    this.ttl
+  });
+
 }
 
+/// TokenDetails is a type containing the token request
+///
+/// response from the REST requestToken endpoint
+///
+/// spec: https://docs.ably.io/client-lib-development-guide/features/#TD1
 class TokenDetails {
-  TokenDetails(this.token);
-  String capability;
-  String clientId;
+
+  /// TD2
```

Comment:
Can we use `Ref: ...` consistently to such ids? (like `/// Ref: TK2a`
`TD2` is rather confusing for someone not familiar with the organization of this project.
`Ref:` is not perfect, but is at least a hint that somewhere else more information can be found.

Reply:
Or even much better 
```dart
/// spec: https://docs.ably.io/client-lib-development-guide/features/#TE1
```

Reply:
@zoechi TD2 is the item right under TD1 in the spec, the class commentary consists of link to TD1, so I felt just mentioning the spec ID would be sufficient. Let me know if it is a better idea to explicitly mention it for each of the fields.

Reply:
I'd add full links everywhere like shown in https://dart.dev/guides/language/effective-dart/documentation#markdown

```
/// spec: https://docs.ably.io/client-lib-development-guide/features/#TE1
```
or
```
/// [spec: TE1](https://docs.ably.io/client-lib-development-guide/features/#TE1)
```
or
```
/// [spec: TE1][https://docs.ably.io/client-lib-development-guide/features/#TE1]
```
Doesn't make much of a difference, but if someone browses API docs in https://pub.dev the links help to navigate directly to the spec

Reply:
a162720d5a628526e5257889ace180d9fbfe0a48